server/src/main/java/org/elasticsearch/common/lucene/RegExp.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.lucene;
+
+import org.apache.lucene.util.automaton.Automaton;
+import org.elasticsearch.core.SuppressForbidden;
+
+/**
+ * Simple wrapper for {@link org.apache.lucene.util.automaton.RegExp} that avoids throwing
+ * StackOverflowErrors when working with regular expressions as this will crash an Elasticsearch node.
+ * Instead, these StackOverflowErrors are turned into IllegalArgumentExceptions.
+ */
+@SuppressForbidden(reason = "catches StackOverflowError")
+public class RegExp {
+
+    private final org.apache.lucene.util.automaton.RegExp wrapped;
+
+    public RegExp(String s) {
+        try {
+            wrapped = new org.apache.lucene.util.automaton.RegExp(s);
+        } catch (StackOverflowError e) {
+            throw new IllegalArgumentException("failed to parse regexp", e);
```

Comment:
Do we really want to propagate `e` here when it might contain a 9000 line stack trace?  In my experience such stack traces are highly recursive and not very useful.

An alternative would be something like:

```suggestion
            throw new IllegalArgumentException("failed to parse regexp due to stack overflow");
```

Reply:
Yeah. We probably do want to pitch the error.

Reply:
Good point!

Reply:
👍

Reply:
@ywelsch Shall we need to print the original regexp string here for further analysis?